src/MOI_wrapper.jl

```diff
@@ -596,31 +598,35 @@ function backward_conic!(model::Optimizer, dA::Array{Float64,2}, db::Array{Float
         z = (x, y - s, [1.0])
         u, v, w = z
 
-        Q = [
+        # reshaping vectors to matrix
+        b = b[:, :]
+        c = c[:, :]
+
+        Q = sparse([
             zeros(n,n)   A'           c;
             -A           zeros(m,m)   b;
             -c'          -b'          zeros(1,1)
-        ]
+        ])
 
         # find gradient of projections on dual of the cones
         Dπv = Dπ([MOI.dual_set(cone) for cone in cones], v)
 
-        M = ((Q - Matrix{Float64}(I, size(Q))) * BlockDiagonal([Matrix{Float64}(I, length(u), length(u)), Dπv, Matrix{Float64}(I,1,1)])) + Matrix{Float64}(I, size(Q))
+        M = ((Q - sparse(1.0I, size(Q))) * blockdiag(sparse(1.0I, length(u), length(u)), Dπv, sparse(1.0I, 1, 1))) + sparse(1.0I, size(Q))
 
         # find projections on dual of the cones
-        πz = vcat([u, π([MOI.dual_set(cone) for cone in cones], v), max.(w,0.0)]...)
+        πz = sparse(vcat([u, π([MOI.dual_set(cone) for cone in cones], v), max.(w,0.0)]...))
 
-        dQ = [
+        dQ = sparse([
             zeros(n,n)    dA'          dc;
             -dA           zeros(m,m)   db;
             -dc'         -db'          zeros(1,1)
-        ]
+        ])
 
         RHS = dQ * πz
         if norm(RHS) <= 1e-4
-            dz = zeros(Float64, size(RHS))
+            dz = sparse(0.0I, size(RHS[:, :]))
```

Comment:
do we need this [:,:] ?

Reply:
yep, again RHS turns out to be a SparseVector

Reply:
removed it